Rocket.Chat/Controllers/Base/MainSplitViewController.swift

```diff
@@ -59,6 +59,22 @@ final class MainSplitViewController: UISplitViewController {
     override var preferredStatusBarStyle: UIStatusBarStyle {
         return view.theme?.appearence.statusBarStyle ?? .default
     }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass {
+            if traitCollection.horizontalSizeClass != .compact {
+                if let nav = primaryViewController as? UINavigationController {
+                    if let subscriptions = nav.viewControllers.first as? SubscriptionsViewController {
+                        if let chat = detailViewController as? ChatViewController {
+                            if let subscription = chat.subscription {
+                                subscriptions.openChat(for: subscription)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Woah 😱. How about using `&&` ?

Reply:
![](https://media.giphy.com/media/px6X1e8dWxdsc/giphy.gif)

Reply:
Oops. Did that for debugging and forgot to optimize.